[3][5] Speedometer HUD, Separate Cameras

nicholassam6425 opened the pull request:
Players now have separate cameras, as well as a simple digital speedometer hud.

nicholassam6425 committed:
speedometer functional on 1 player

nicholassam6425 committed:
separate scripts

nicholassam6425 committed:
seperated cameras per player

nicholassam6425 committed:
remove debug line

Sameer1223 commented on Assets/Scripts/UI/CameraController.cs:
```diff
@@ -0,0 +1,16 @@
+    private GameObject cameraHolder;
+    public override void OnNetworkSpawn()
+    {
+        cameraHolder = transform.GetChild(2).gameObject;
```
Generally want to avoid runtime lookups with an integer or string. You can use [SerializeField] and then drag in the element in the editor you need to store reference. This way if we ever change the structure of the prefab we don't get random bugs in code.

`[SerializeField]`
`private ClassName variableName;`

And all you need to do is drag in the game object in the editor into the script component on the object.
Then you can reference the variableName.

Sameer1223 commented on Assets/Scripts/UI/PlayerUI.cs:
```diff
@@ -0,0 +1,23 @@
+    }
+    private void UpdateUI() {
+        currentSpeed = rb.linearVelocity.magnitude * 2.237f;
+        speedometerText.text = currentSpeed.ToString("N0") + " mph";
```
Same thing here as mentioned above

Sameer1223 commented on Assets/Scripts/UI/PlayerUI.cs:
```diff
@@ -0,0 +1,23 @@
+        speedometerText = transform.GetChild(1).GetChild(0).GetComponent<TMP_Text>();
+    }
+    private void UpdateUI() {
+        currentSpeed = rb.linearVelocity.magnitude * 2.237f;
```
What's the constant for? Is it to convert to Mph?
Also add any constants at the top of the file as a variable

Sameer1223 commented:
Also can we make it so that even if I resize or play on a different aspect ratio the text shows

nicholassam6425 committed:
serializefields, constants, overlay ui, and aspect ratios

use SerializeField rather than object lookups, name constants, make ui overlay instead of in-world object, and support different aspect ratios for the ui

Sameer1223 reviewed (approved):
LGTM
